### PR TITLE
add map editor page

### PR DESCRIPTION
**Pull request by light5551**

### CODE REVIEW

File: book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+</div>
+
+## How to create new map?
+
+You should specify size(width, height) of map and size of tile. Default tile size: **0.585**
+<div figure-id="fig:editor2" figure-caption="How to create new map">
+  <img src="images/create_map.png" style='width: 25em; height:auto'/>
+</div>
+
+Result of generation map
```
Comment (yosoufe) [inline book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md:30]: `map generation` or `generation of the map`?

File: book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+<div figure-id="fig:editor3" figure-caption="new map">
+  <img src="images/result_of_creation_map.png" style='width: 25em; height:auto'/>
+</div>
+
+## How to add an object to the map
+ 
+You can add any object from left menu. Except first two block, which describe type of tile.
```
Comment (yosoufe) [inline book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md:38]: `two blocks`, plural `s` is missing?

File: book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+<div figure-id="fig:editor6" figure-caption="example of map">
+  <img src="images/tile_road_ex.png" style='width: 25em; height:auto'/>
+</div>
+
+<div figure-id="fig:editor7" figure-caption="choose tiles">
+  <img src="images/selection_tiles.png" style='width: 25em; height:auto'/>
+</div>
+
+## Rotate tiles
+For rotation you should select required tiles and click button of rotation(see fig. 9)
```
Comment (jasonhu5) [inline book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md:63]:
```suggestion
For rotation you should select required tiles and click the rotation button in the tool bar (Fig. 8). See Fig. 9 for the illustrated outcome.
```

File: book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+<div figure-id="fig:editor3" figure-caption="add object to map">
+  <img src="images/add_stop.png" style='width: 25em; height:auto'/>
+</div>
+
+For editing attributes of object you should do right click of object and change attributes in opened window. Window for editing object:
+<div figure-id="fig:editor4" figure-caption="change attribute of obj">
+  <img src="images/edit_attr.png" style='width: 25em; height:auto'/>
+</div>
+
+For editing tiles, you should activate brush mode(see Fig.5). After that, you can choose type of tile in left menu.  After that, a should select tiles.
```
Comment (jasonhu5) [inline book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md:49]:
```suggestion
For editing tiles, you should activate the brush mode (see Fig.5). After that, you should choose the type of tiles in left menu. Then, one could select/brush the grids to apply the selected tile type.
```

File: book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+## How to add an object to the map
+ 
+You can add any object from left menu. Except first two block, which describe type of tile.
+For adding object you should do double click object.
+<div figure-id="fig:editor3" figure-caption="add object to map">
+  <img src="images/add_stop.png" style='width: 25em; height:auto'/>
+</div>
+
+For editing attributes of object you should do right click of object and change attributes in opened window. Window for editing object:
```
Comment (jasonhu5) [inline book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md:44]:
```suggestion
For editing attributes of an object you should do right click of the object and edit its attributes in the opened window. The window should look like:
```

File: book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+Result of generation map
+
+<div figure-id="fig:editor3" figure-caption="new map">
+  <img src="images/result_of_creation_map.png" style='width: 25em; height:auto'/>
+</div>
+
+## How to add an object to the map
+ 
+You can add any object from left menu. Except first two block, which describe type of tile.
+For adding object you should do double click object.
```
Comment (jasonhu5) [inline book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md:39]:
```suggestion
For adding an object, you should double click the object in the left menu.
```

File: book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md (hunk 2 above)
Comment (jasonhu5) [reply book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md:38]:
```suggestion
You can add any object(s) from left menu except the first two blocks, "Road tiles" and "Fill tiles", which describe the types of tiles.
```

File: book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+<div class='requirements' markdown="1">
+
+Requires: Knowledge about what a Autolab is [](#part:autolab-definition)
+
+Results: generated map for duckietown
+</div>
+
+## Map editor
+
+You can launch map editor, using command:
```
Comment (jasonhu5) [inline book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md:13]:
```suggestion
You can launch the map editor, using command:
```

File: book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+    laptop $ dts map editor
+
+You will see:
+
+<div figure-id="fig:editor1" figure-caption="Map editor">
+  <img src="images/simple_editor.png" style='width: 25em; height:auto'/>
+</div>
+
+## How to create new map?
```
Comment (jasonhu5) [inline book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md:23]:
```suggestion
## How to create a new map?
```

File: book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+
+You should specify size(width, height) of map and size of tile. Default tile size: **0.585**
```
Comment (jasonhu5) [inline book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md:25]:
```suggestion
You should specify sizes(width, height) of the map and the size of tiles. Default tile size: **0.585**
```

File: book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md (hunk 1 above)
Comment (jasonhu5) [reply book/opmanual_autolab/15_autolab_minimal_requirements/40_autolab_map_editor.md:30]:
```suggestion
Then the map generation process should be initialized. Sample editor window:
```

**Review by jasonhu5 — CHANGES_REQUESTED** [review]
Mostly grammar minors.

**Review by jasonhu5 — APPROVED** [review]
looks good to me

**Pull request merged**